### PR TITLE
Add OpenAPI3 strict references option to Committee (#343)

### PR DESCRIPTION
**Pull request by makdad**
@ota42y please review my first implementation attempt to handle #343.

Issues I noticed while implementing:

* This would force the `openapi_parser` Gem version to 1.0.0.beta, since that Gem is merely in beta, you may not be comfortable releasing this change on Committee mainline yet?

* Do we need to provide a similar functionality for Hyper-schema?  I personally don't have any need for it in my work, but I can understand how it's nice if Committee can behave similarly for different schema types.

* If this can fit into 5.x before it is released, then we can likely turn on `true` as a default from 6.x.  I am wondering - is there any use case to even allow it to be configured as false?  I guess after releasing this version, we may hear from users who want it to stay configurable.

Also, similar to openapi_parser, I added `.ruby-version` and added `.idea` to `.gitignore` for RubyMine (JetBrains IDE).

### CODE REVIEW

**Review by ota42y — APPROVED** [review]
> you may not be comfortable releasing this change on Committee mainline yet?

I think it's no problem :)

> Do we need to provide a similar functionality for Hyper-schema?

We don't need to add it, some features are already available only in OpenAPI 3.
If anyone wants it, we should still work on another PR.

>If this can fit into 5.x before it is released, then we can likely turn on true as a default from 6.x

🙆‍♀️ 🙆‍♀️ 🙆‍♀️ 🙆‍♀️ 

I think this change is OK!
But master branch is broken and I fixed it, so please rebase newest master 🙇

File: lib/committee/middleware/base.rb [hunk 1]
```diff
@@ -30,11 +30,12 @@ def call(env)
       class << self
         def get_schema(options)
           schema = options[:schema]
-          unless schema
-            schema = Committee::Drivers::load_from_file(options[:schema_path]) if options[:schema_path]
-
-            raise(ArgumentError, "Committee: need option `schema` or `schema_path`") unless schema
+          if !schema && options[:schema_path]
+            # In the future, we could have `parser_options` as an exposed config?
```
[the remark this answers is not in the document]
Comment (makdad) [reply lib/committee/middleware/base.rb:34]: @ota42y Thanks for the feedback, I think I generally understand your direction.  Is that something we should open a separate GitHub issue for and then think of an architecture there?

**Comment by makdad** [comment]
@ota42y I changed the commit name as well as rebased off of master.

File: lib/committee/middleware/base.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ota42y) [reply lib/committee/middleware/base.rb:34]: Yes, and I created new issue 🙋 
https://github.com/interagent/committee/issues/347

**Comment by ota42y** [comment]
Thanks!!!!!

**Comment by ota42y** [comment]
I released 5.0.0 with this changes!
https://rubygems.org/gems/committee/versions/5.0.0

**Pull request merged**